src/Compilers/Core/Portable/CommandLine/SarifV2ErrorLogger.cs

```diff
@@ -167,16 +167,19 @@ private void WriteLocations(Location location, IReadOnlyList<Location> additiona
             }
         }
 
-        protected override void WritePhysicalLocation(Location diagnosticLocation)
+        protected override void WritePhysicalLocation(Location diagnosticLocation, SourceReferenceResolver? resolver)
         {
             Debug.Assert(HasPath(diagnosticLocation));
 
             FileLinePositionSpan span = diagnosticLocation.GetLineSpan();
+            var path = diagnosticLocation.SourceTree is { } tree
+                ? tree.GetDisplayPath(diagnosticLocation.SourceSpan, resolver)
```

Comment:
This also handles line directives. I didn't add a new test here, but I do have another PR in the works, which is maybe better to target main with, in order to make adding/maintaining the tests easier in this area.